Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule.py

```diff
@@ -3565,3 +3626,282 @@ def remove_tag_from_endpoints_command(client: CoreClient, args: Dict):
+    return {
+        "Role Name": role_data.get("pretty_name"),
+        "Description": role_data.get("description"),
+        "Permission": role_data.get("permissions", []),
+        "Users": role_data.get("users", []),
+        "Groups": role_data.get("groups", []),
+    }
+
+
+def find_the_cause_error(err: Exception) -> str:
+    """
+    Finds the cause of an error message related to a missing ID, group or role.
+
+    Args:
+        err: A string containing the error message to parse.
+
+    Returns:
+        A string describing the cause of the error message.
+
+    """
+    pattern = r"(id |Group |Role )\\?'([/A-Za-z 0-9]+)\\?'"
+    if match := re.search(pattern, str(err)):
+        return f'Error: {match[1]}{match[2]} was not found'
+    return "Error: "
+
+
+def handle_error(e, type_: str | None, custom_msg: str | None) -> None:
+    """
+    Handles an error that may occur during a process.
+
+    Args:
+        e (Exception): The error that occurred.
+        type (str | None): The type of resource associated with the error(Role id or Group), if applicable.
+        custom_msg (str | None): A custom error message to be included in the raised ValueError, if desired.
+
+    Raises:
+        ValueError: If the error message indicates that the resource was not found, a more detailed error message
+            is constructed using the `find_the_cause_error` function and raised with the original error as the cause.
+    """
+
+    if (
+        e.res is not None
+        and e.res.status_code == 500
+        and 'was not found' in str(e)
+    ):
+        error_msg = find_the_cause_error(e)
+        raise ValueError(
+            f'{error_msg}{custom_msg if type_ in ("Group", "Role") else ""}. Full error message: {e}'
+        ) from e
+
+
+def get_list_users_command(client: CoreClient, args: dict[str, str]) -> CommandResults:
+    """
+    Returns a list of all users using the Core API client.
+
+    Args:
+        client: A CoreClient instance used for connecting to the Core API.
+        args: A dictionary containing additional arguments. Possible keys include:
+            - integration_context_brand (str): The name of the integration context brand.
+
+    Returns:
+        A CommandResults object containing the readable_output and outputs fields.
+
+    Raises:
+        ValueError: If the API connection failed.
+    """
+
+    def parse_user(user: dict[str, Any]) -> dict[str, Any]:
+        return {
+            'User email': user.get('user_email'),
+            'First Name': user.get('user_first_name'),
+            'Last Name': user.get('user_last_name'),
+            'Role': user.get('role_name'),
+            'Type': user.get('user_type'),
+            'Groups': user.get('groups'),
+        }
+
+    try:
+        listed_users: list[dict[str, Any]] = client.get_list_users().get('reply', [])
+    except Exception as e:
+        raise ValueError(f'API connection failed: {e}') from e
```

Comment:
```suggestion
    try:
        listed_users: list[dict[str, Any]] = client.get_list_users().get('reply', [])
    except Exception as e:
        raise ValueError(f'API connection failed: {e}') from e
```
1. Do we _need_ this `try`? What does it look like otherwise? 
2. Is the error always a connection error? What if there's an internal server error? 

(This comment applies to the rest of the commands using this try-except structure)